tests/test_s3_select.py

```diff
@@ -0,0 +1,161 @@
+import logging
+
+import pandas as pd
+import pytest
+
+import awswrangler as wr
+
+logging.getLogger("awswrangler").setLevel(logging.DEBUG)
+
+
+@pytest.mark.parametrize("use_threads", [True, False, 2])
+def test_full_table(path, use_threads):
+    df = pd.DataFrame({"c0": [1, 2, 3], "c1": ["foo", "boo", "bar"], "c2": [4.0, 5.0, 6.0]})
+
+    # Parquet
+    file_path = f"{path}test_parquet_file.snappy.parquet"
+    wr.s3.to_parquet(df, file_path, compression="snappy")
+    df2 = wr.s3.select_query(
+        sql="select * from s3object",
+        path=file_path,
+        input_serialization="Parquet",
+        input_serialization_params={},
+        use_threads=use_threads,
+    )
+    assert df.equals(df2)
+
+    # CSV
+    file_path = f"{path}test_csv_file.csv"
+    wr.s3.to_csv(df, file_path, index=False)
+    df3 = wr.s3.select_query(
+        sql="select * from s3object",
+        path=file_path,
+        input_serialization="CSV",
+        input_serialization_params={"FileHeaderInfo": "Use", "RecordDelimiter": "\n"},
+        use_threads=use_threads,
+    )
+    assert len(df.index) == len(df3.index)
+    assert list(df.columns) == list(df3.columns)
+    assert df.shape == df3.shape
```

Comment:
When reading from a CSV, the S3 Select Engine returns everything as a string (compared to Parquet and JSON where types are preserved). It means we cannot perfectly compare the two dataframes